### ISSUE TITLE
Add config option that allows admins to disable the Basic authentication scheme

### ISSUE THREAD

**Issue by korydraughn** [irods/irods_client_http_api#387]
## Enhancement

The HTTP API always allows users to auth via the Basic auth scheme. Admins may only want to allow auth via the OIDC flow. For that to be possible, the HTTP API must expose an option that lets the admin disable the Basic auth scheme.

**Comment by MartinFlores751** [comment]
Thinking out loud, would the provided option be something similar to `disable_basic_auth: true` or something like:

```json
{
  "enabled_auth_schemes": [ "basic", "openid" ]
}
```

Put differently, do we want "exclude" or "include" based options?

**Comment by trel** [comment]
i like 'positive' options, rather than 'negative'... in general.

**Comment by korydraughn** [comment]
Yes, what you're proposing is a good first start. That design covers all cases IMO.

An empty list would not be allowed and result in a failure during server startup.

**Comment by korydraughn** [comment]
> What advantages does this bring practically?

The advantage is that the admin gets to harden their deployment a little more.

Instead of an enum, basic auth could be treated like oidc. That is, if basic is defined in the config file, then it's enabled. If it's not defined, then it's disabled.

> What if `authentication` took only one auth scheme stanza and refuse to start the server otherwise?

Are you saying that the json schema only allow basic or oidc to be defined. If none or both config stanzas are defined, then the server treats that as an error and doesn't start?

**Comment by korydraughn** [comment]
That's a fair assessment. However, we don't know how people are using the software. It's possible no one needs basic and oidc enabled simultaneously. It's also possible that there are use-cases for having both enabled.

No decisions have been made yet, so it's very possible that what you've proposed is what gets implemented.
